pkg/kubelet/kubelet_getters.go

```diff
@@ -237,12 +238,20 @@ func (kl *Kubelet) GetNode() (*v1.Node, error) {
 // in which case return a manufactured nodeInfo representing a node with no pods,
 // zero capacity, and the default labels.
 func (kl *Kubelet) getNodeAnyWay() (*v1.Node, error) {
+	var node *v1.Node
+	var err error
 	if kl.kubeClient != nil {
-		if n, err := kl.nodeLister.Get(string(kl.nodeName)); err == nil {
-			return n, nil
+		if node, err = kl.nodeLister.Get(string(kl.nodeName)); err == nil {
```

Comment:
Is it possible to just block here until the node cache gets populated? And exit kubelet after a period of time as kubelet should not continue until it gets up to date node info.

Reply:
I had the same thought when looking at this code.